app/forms/questionnaire_form.py

```diff
@@ -32,24 +30,21 @@ def __init__(self, schema, block_json, answer_store, metadata, **kwargs):
 
     def validate(self):
         """
-        Validate this form as usual and check for any form-level date-range/calculated validation errors
-        :return:
+        Validate this form as usual and check for any form-level validation errors based on question type
+        :return: boolean
         """
         valid_fields = FlaskForm.validate(self)
-        valid_date_range_form = True
-        valid_calculated_form = True
+        valid_form = True
 
         for question in self.schema.get_questions_for_block(self.block_json):
             if question['type'] == 'DateRange':
-                valid_date_range = self.validate_date_range_question(question)
-                if valid_date_range_form:
-                    valid_date_range_form = valid_date_range
+                valid_form = self.validate_date_range_question(question)
             elif question['type'] == 'Calculated':
-                valid_calculated = self.validate_calculated_question(question)
-                if valid_calculated_form:
-                    valid_calculated_form = valid_calculated
+                valid_form = self.validate_calculated_question(question)
+            elif question['type'] == 'MutuallyExclusive' and valid_fields:
+                valid_form = self.validate_mutually_exclusive_question(question)
```

Comment:
If we have multiple questions on a page then the last valid_form value will be the only one that is used. If one of the earlier valid_forms is a failure it will not be picked up.

Reply:
Ah makes sense. Will change. Good spot 👍